### PR TITLE
Setup Static site in EC2 server

### PR DESCRIPTION
**Pull request by anish3389**

### CODE REVIEW

File: .github/workflows/main.yaml [hunk 1]
```diff
@@ -0,0 +1,96 @@
+name: Provision Terraform Resources
+
+on:
+  push:
+    branches: [ INTERN-679-PIPELINE ]
```
Comment (mukes137) [inline .github/workflows/main.yaml:5]: use lowercase letters for branch name.

File: .github/workflows/main.yaml [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        with:
+          aws-access-key-id: ${{ env.awsAccessKeyId }}
+          aws-secret-access-key: ${{ env.awsSecretAccessKey }}
+          aws-session-token: ${{ env.awsSessionToken }}
+          aws-region: ${{ env.awsRegion }}
+
+      - name: Terraform Initialize
+        id: init
+        run: cd terraform && terraform init -backend-config="bucket=${{ env.backendS3Bucket }}" -backend-config="key=${{ env.backendS3AccountId }}/${{ env.backendS3StateFile }}" -backend-config="region=${{ env.backendS3Region }}"
+
```
Comment (mukes137) [inline .github/workflows/main.yaml:45]: Also add the plan phase for terraform.
If terrform plan succeeds then only move to terraform apply.

File: .github/workflows/main.yaml [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        run: |
+          cd terraform && terraform apply -var-file=dev.tfvars -auto-approve -no-color -target=${{ env.tfPrerequisiteResource }}
+
+  upload_code_to_bucket:
+    needs: [ create_s3 ]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: configure aws credentials
```
Comment (mukes137) [inline .github/workflows/main.yaml:58]: Why are we configuring aws creds twice.

File: terraform/variables.tf [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+}
+
+variable "silo" {
+  description = "Name to be used on all the resources as identifier"
+  type        = string
+  default     = ""
+}
+
+
+variable "ami_id" {}
```
Comment (mukes137) [inline terraform/variables.tf]: please add variables type and description for all variables.

**Review by mukes137 — CHANGES_REQUESTED** [review]

File: .github/workflows/main.yaml (hunk 1 above)
Comment (anish3389) [reply .github/workflows/main.yaml:5]: Resolved

File: terraform/variables.tf (hunk 4 above)
Comment (anish3389) [reply terraform/variables.tf]: Updated the code

File: .github/workflows/main.yaml (hunk 2 above)
Comment (anish3389) [reply .github/workflows/main.yaml:45]: Added plan

**Review by mukes137 — APPROVED** [review]

**Pull request merged**